File: wk5/java/romanToInteger.java
// problem link: https://leetcode.com/problems/roman-to-integer/description/

class Solution {
    public int romanToInt(String s) {
        int ret = 0;
        for(int i = 0; i < s.length(); i++) {
            switch(s.charAt(i)) {
                case 'V':
                    ret += 5;
                    break;
                case 'L':
                    ret += 50;
                    break;
                case 'D':
                    ret += 500;
                    break;
                case 'M':
                    ret += 1000;
                    break;
                case 'I':
                    if(i + 1 < s.length() && (s.charAt(i + 1) == 'V' || s.charAt(i + 1) == 'X'))
                        ret -= 1;
                    else
                        ret += 1;
                    break;
                case 'X':
                    if(i + 1 < s.length() && (s.charAt(i + 1) == 'L' || s.charAt(i + 1) == 'C'))
                        ret -= 10;
                    else
                        ret += 10;
                    break;
                case 'C':
                    if(i + 1 < s.length() && (s.charAt(i + 1) == 'D' || s.charAt(i + 1) == 'M'))
                        ret -= 100;
                    else
                        ret += 100;
                    break;
            }
        }
        return ret;
    }
}

File: wk5/py/romantointeger.py
#link: https://leetcode.com/problems/roman-to-integer/

class Solution:
    def romanToInt(self, s: str) -> int:
        values = {"I":1,"V":5,"X":10,"L":50,"C":100,"D":500,"M":1000}
        total = 0
        for i in range(len(s)-1):
            char, nextChar = s[i], s[i+1]

            if values[char] >= values[nextChar]:
                total += values[char]
            else:
                total -= values[char]
        
        return total + values[s[-1]]